modules/contact/src/actions/ContactAction.C

```diff
@@ -179,6 +186,105 @@ ContactAction::act()
         addDiracContact();
     }
   }
+
+  if (_current_task == "add_aux_kernel")
+  { // Add ContactPenetrationAuxAction.
+    if (!_problem->getDisplacedProblem())
+    {
+      mooseError("Contact requires updated coordinates.  Use the 'displacements = ...' line in the "
+                 "Mesh block.");
+    }
+
+    {
+      InputParameters params = _factory.getValidParams("PenetrationAux");
+      params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_LINEAR};
+
+      // Extract global params
+      if (isParamValid("parser_syntax"))
+        _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
+
+      params.set<std::vector<BoundaryName>>("boundary") = {_slave};
+      params.set<BoundaryName>("paired_boundary") = _master;
+      params.set<AuxVariableName>("variable") = "penetration";
+      params.set<MooseEnum>("order") = _order;
+
+      params.set<bool>("use_displaced_mesh") = true;
+      std::stringstream name;
```

Comment:
`std::string name = _name + "_contact_" + counter++;`

Doesn't something like that work?

Reply:
Addressed it in the last commit.